Tasks/Common/utility-common/telemetry.ts

```diff
@@ -0,0 +1,66 @@
+                'SYSTEM_PLANID': tl.getVariable('SYSTEM_PLANID'),
+                'SYSTEM_COLLECTIONID': tl.getVariable('SYSTEM_COLLECTIONID'),
+                'AGENT_BUILDDIRECTORY': tl.getVariable('AGENT_BUILDDIRECTORY'),
+                'AGENT_HOMEDIRECTORY': tl.getVariable('AGENT_HOMEDIRECTORY'),
+                'AGENT_WORKFOLDER': tl.getVariable('AGENT_WORKFOLDER'),
+                'AGENT_ROOTDIRECTORY': tl.getVariable('AGENT_ROOTDIRECTORY'),
+                'AGENT_TOOLSDIRECTORY': tl.getVariable('AGENT_TOOLSDIRECTORY'),
+                'AGENT_SERVEROMDIRECTORY': tl.getVariable('AGENT_SERVEROMDIRECTORY'),
+                'AGENT_TEMPDIRECTORY': tl.getVariable('AGENT_TEMPDIRECTORY'),
+                'AGENT_ID': tl.getVariable('AGENT_ID'),
+                'AGENT_MACHINENAME': tl.getVariable('AGENT_MACHINENAME'),
+                'AGENT_NAME': tl.getVariable('AGENT_NAME'),
+                'AGENT_JOBSTATUS': tl.getVariable('AGENT_JOBSTATUS'),
+                'AGENT_OS': tl.getVariable('AGENT_OS'),
+                'AGENT_VERSION': tl.getVariable('AGENT_VERSION'),
+                'BUILD_ARTIFACTSTAGINGDIRECTORY': tl.getVariable('BUILD_ARTIFACTSTAGINGDIRECTORY'),
+                'BUILD_BINARIESDIRECTORY': tl.getVariable('BUILD_BINARIESDIRECTORY'),
+                'BUILD_BUILDID': tl.getVariable('BUILD_BUILDID'),
+                'BUILD_BUILDNUMBER': tl.getVariable('BUILD_BUILDNUMBER'),
+                'BUILD_BUILDURI': tl.getVariable('BUILD_BUILDURI'),
+                'BUILD_CONTAINERID': tl.getVariable('BUILD_CONTAINERID'),
+                'BUILD_DEFINITIONNAME': tl.getVariable('BUILD_DEFINITIONNAME'),
+                'BUILD_DEFINITIONVERSION': tl.getVariable('BUILD_DEFINITIONVERSION'),
+                'BUILD_REASON': tl.getVariable('BUILD_REASON'),
+                'BUILD_REPOSITORY_CLEAN': tl.getVariable('BUILD_REPOSITORY_CLEAN'),
+                'BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT': tl.getVariable('BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT'),
+                'BUILD_REPOSITORY_ID': tl.getVariable('BUILD_REPOSITORY_ID'),
+                'BUILD_REPOSITORY_LOCALPATH': tl.getVariable('BUILD_REPOSITORY_LOCALPATH'),
+                'BUILD_REPOSITORY_NAME': tl.getVariable('BUILD_REPOSITORY_NAME'),
+                'BUILD_REPOSITORY_PROVIDER': tl.getVariable('BUILD_REPOSITORY_PROVIDER'),
+                'BUILD_REPOSITORY_URI': tl.getVariable('BUILD_REPOSITORY_URI'),
+                'BUILD_SOURCEBRANCH': tl.getVariable('BUILD_SOURCEBRANCH'),
+                'BUILD_SOURCEBRANCHNAME': tl.getVariable('BUILD_SOURCEBRANCHNAME'),
+                'BUILD_SOURCESDIRECTORY': tl.getVariable('BUILD_SOURCESDIRECTORY'),
+                'BUILD_SOURCEVERSION': tl.getVariable('BUILD_SOURCEVERSION'),
+                'BUILD_STAGINGDIRECTORY': tl.getVariable('BUILD_STAGINGDIRECTORY'),
+                'agent.proxyurl': tl.getVariable("agent.proxyurl")
+            });
+            let copy = Object.assign(JSON.parse(commonTelem), telem);
```

Comment:
JSON.parse shouldn't be needed if you don't stringify commonTelem first.

Reply:
Fixed